### ISSUE TITLE
Algo: secant (2 members)

### ISSUE THREAD

**Issue by laserlab** [ubsuny/CP1-24-HW7#6]
Each algorithm has to be implemented in three ways.

wrapper for scipy implementation (due Wednesday)
pure python implementation (due Friday)
c/c++ implementation using ctypes in python (due Monday)

**Comment by iglesias-cardinale** [comment]
I'll work on this one

**Comment by avgagliardo** [comment]
@iglesias-cardinale any preferences on how you want to split up the task?

**Comment by avgagliardo** [comment]
I think the scipy implementation should be very lightweight so if you want to take care of the first two I'll take care of the ctypes implementation. Or we can flip it if you prefer, up to you.

**Comment by iglesias-cardinale** [comment]
@avgagliardo yeah that sounds good to me. I'll do a PR for the scipy wrapper either late tonight or early tomorrow.

**Comment by iglesias-cardinale** [comment]
@avgagliardo I added a couple of unit tests in PR #29, but I figure that doing the c/c++ implementation will be pretty non-trivial where the other two methods are pretty easy, so I figured this would make the work more 50/50. That way you only have to write unit tests for your c++ function and you can still get two PRs if you break them up.

**Comment by avgagliardo** [comment]
> @avgagliardo I added a couple of unit tests in PR #29, but I figure that doing the c/c++ implementation will be pretty non-trivial where the other two methods are pretty easy, so I figured this would make the workload more 50/50. That way you only have to write unit tests for your c++ function and you can still get two PRs if you break them up.

Much appreciated! Hoping to get my PRs up tonight or tomorrow but I'll ping you on one if you need a review still.

**Comment by avgagliardo** [comment]
> @avgagliardo I added a couple of unit tests in PR #29, but I figure that doing the c/c++ implementation will be pretty non-trivial where the other two methods are pretty easy, so I figured this would make the workload more 50/50. That way you only have to write unit tests for your c++ function and you can still get two PRs if you break them up.

Just pushed the C++ implementation
